core/translations/fr.json

```diff
@@ -52,52 +52,52 @@
     "address__title_yours": "La vôtre",
     "address_details__account_info": "Infos sur le compte",
     "address_details__derivation_path": "Chemin de dérivation",
-    "address_details__derivation_path_colon": "Chemin de dérivation :",
+    "address_details__derivation_path_colon": "Chemin de dérivation:",
     "address_details__title_receive_address": "Adr. de récep.",
     "address_details__title_receiving_to": "Récep. sur",
-    "authenticate__confirm_template": "Autoriser l'ord. connecté à conf. que votre {0} est auth. ?",
+    "authenticate__confirm_template": "Autoriser l'ord. connecté à conf. que votre {0} est auth.?",
```

Comment:
Do we really want to do this for `?` too even if it doesn't buy us anything?

Btw [wikipedia](https://en.wikipedia.org/wiki/Punctuation#Other_languages) claims this about canadian french:
> In the [French](https://en.wikipedia.org/wiki/French_language) of [France](https://en.wikipedia.org/wiki/France) and [Belgium](https://en.wikipedia.org/wiki/Belgium), the marks ⟨:⟩, ⟨;⟩, ⟨?⟩ and ⟨!⟩ are preceded by a thin space. In [Canadian French](https://en.wikipedia.org/wiki/Canadian_French), this is only the case for ⟨:⟩.

Reply:
I think it can be argued both ways. What it buys us is consistency with other languages, which was the whole point of this PR. Although I also agree that in the case of `?` it might not matter, since we will probably not have cases of strings with or without `?` that would need to be added from code?

Anyway, your decision. If you think I should keep the space in this case, I can definitely go that way. I don't have a strong preference for either way...

Reply:
Code consistency is nice but I suppose user experience is the most important. French people are proud of their culture and we need to do everything we can to compete with Ledger :sweat_smile: 

> Although I also agree that in the case of ? it might not matter, since we will probably not have cases of strings with or without ? that would need to be added from code?

I think that's the distinction, with `:` strings are often constructed dynamically (maybe even mixed with others that already have `:` at the end) while `?` and `!` are often at the end of whole sentence. So my suggestion is to use the `:` handling from this PR but keep the spaces before `?` and `!` in fr.json.

Reply:
> So my suggestion is to use the `:` handling from this PR but keep the spaces before `?` and `!` in fr.json.

https://github.com/trezor/trezor-firmware/pull/4118/commits/c4665b8a0e82f2e8761e339e9aa88de399d7fb27